File: 1399-Count-Largest-Group/1399.c
int countLargestGroup(int n) {
    short countarray[36] = {}; //As 36 is largest digit sum possible (1 <= n <= 10^4)
    int returnCounter = 0;
    short mostFrequentCount = 0;

    short tnum = 0;
    short digit = 0;
    short digitSum = 0;
    for(short num=1; num <= n; num++) {
        tnum = num; 
        while(tnum != 0) {
            digit = tnum % 10; 
            digitSum += digit;
            tnum /= 10;
        }

        countarray[digitSum-1]++;
        digitSum = digit = 0; 
    }

    for(char i=0; i<36; i++) {
        if(countarray[i] > mostFrequentCount) {
            returnCounter = 1; 
            mostFrequentCount = countarray[i];
        } else if(countarray[i] == mostFrequentCount) {
            returnCounter++;
        }
    }

    return returnCounter;
}

File: 1399-Count-Largest-Group/1399.rs
impl Solution {
    pub fn count_largest_group(n: i32) -> i32 {
    let mut countarray = [0; 36]; // As 36 is the largest digit sum possible (1 <= n <= 10^4)
    let mut returnCounter = 0;
    let mut mostFrequentCount = 0;

    let mut tnum: i16;
    let mut digit: i16;
    let mut digitSum: i16;

    for num in 1..=n {
        tnum = num as i16;
        digitSum = 0;

        while tnum != 0 {
            digit = tnum % 10;
            digitSum += digit;
            tnum /= 10;
        }

        countarray[(digitSum - 1) as usize] += 1;
    }

    for count in countarray.iter() {
        if *count > mostFrequentCount {
            returnCounter = 1;
            mostFrequentCount = *count;
        } else if *count == mostFrequentCount {
            returnCounter += 1;
        }
    }

    returnCounter
    }
}
